test/Elastic.Apm.Tests/SpanTests.cs

```diff
@@ -0,0 +1,32 @@
+using Elastic.Apm.Model;
+using Elastic.Apm.Tests.Mocks;
+using Xunit;
+
+namespace Elastic.Apm.Tests
+{
+	public class SpanTests
+	{
+		[Fact]
+		public void End_ShouldRestoreParentSpan_WhenTransactionIsNotSampled()
+		{
+			// Arrange
+			var currentExecutionSegmentsContainer = new NoopCurrentExecutionSegmentsContainer();
+			var payloadSender = new NoopPayloadSender();
+			var logger = new NoopLogger();
+			var configurationReader = new MockConfigSnapshot();
+
+			var transaction = new Transaction(logger, "transaction", "type", new Sampler(1.0), null, payloadSender, configurationReader,

```

Comment:
I assume it should be `new Sampler(0)` (or [`config: new MockConfigSnapshot(transactionSampleRate: "0")`](https://github.com/elastic/apm-agent-dotnet/blob/480cb4fdcd5b2034b62ed6b0fb9d5da6026ff3e7/test/Elastic.Apm.Tests/SpanTests.cs#L64) if we use only public API) because otherwise the test passes even without the fix to production code.